client/__tests__/e2e/cellxgeneActions.ts

```diff
@@ -43,262 +37,256 @@ export async function drag(testId: any, start: any, end: any, lasso = false) {
-    allHistograms.map((hist) =>
-      page.evaluate((elem) => elem.dataset.testid, hist)
-    )
-  );
-
-  return testIDs.map((id) => id.replace(/^histogram-/, ""));
-}
-
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function getAllCategoriesAndCounts(category: any) {
+export async function getAllCategoriesAndCounts(
+  category: string,
+  page: Page
+): Promise<[string, string][]> {
   // these load asynchronously, so we have to wait for the specific category.
-  await waitByID(`category-${category}`);
-
-  return page.$$eval(
-    `[data-testid="category-${category}"] [data-testclass='categorical-row']`,
-    (rows) =>
-      Object.fromEntries(
-        rows.map((row) => {
-          // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-          const cat = row
-            .querySelector("[data-testclass='categorical-value']")
-            .getAttribute("aria-label");
-
-          const count = (
-            row.querySelector(
-              "[data-testclass='categorical-value-count']"
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-            ) as any
-          ).innerText;
-
-          return [cat, count];
-        })
-      )
+  const categoryRows = await page
+    .getByTestId(`category-${category}`)
+    .getByTestId("categorical-row")
+    .all();
+
+  return Object.fromEntries(
+    await Promise.all(
+      categoryRows.map(async (row) => {
+        const cat =
+          (await row
+            .getByTestId("categorical-value")
+            .getAttribute("aria-label")) ?? "";
+
+        const count = await row
+          .getByTestId("categorical-value-count")
+          .innerText();
+
+        return [cat, count];
+      })
+    )
   );
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function getCellSetCount(num: any) {
-  await clickOn(`cellset-button-${num}`);
-  return getOneElementInnerText(`[data-testid='cellset-count-${num}']`);
+export async function getCellSetCount(
+  num: number,
+  page: Page
+): Promise<string> {
+  await page.getByTestId(`cellset-button-${num}`).click({ force: true });
+  return page.getByTestId(`cellset-count-${num}`).innerText();
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function resetCategory(category: any) {
+export async function resetCategory(
+  category: string,
+  page: Page
+): Promise<void> {
   const checkboxId = `${category}:category-select`;
-  await waitByID(checkboxId);
-  const checkedPseudoclass = await page.$eval(
+  await page.getByTestId(checkboxId);
```

Comment:
this doesn't do anything, since locator doesn't return a promise

If we wanna wait for the element to be visible, maybe:

```ts
await page.getByTestId(checkboxId).waitFor({ state: "visible" });
```